influxdb3_write/src/paths.rs

```diff
@@ -0,0 +1,128 @@
+use crate::SegmentId;
+use std::convert::AsRef;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// File extension for catalog files
+const CATALOG_FILE_EXTENSION: &str = "json";
+
+/// File extension for parquet files
+const PARQUET_FILE_EXTENSION: &str = "parquet";
+
+/// File extension for segment files
+const SEGMENT_FILE_EXTENSION: &str = "wal";
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CatalogFilePath(PathBuf);
+
+impl CatalogFilePath {
+    pub fn new(prefix: impl Into<PathBuf>, sequence_number: u64) -> Self {
+        let mut path = prefix.into();
+        path.push("catalogs");
+        path.push(format!("{sequence_number:010}"));
+        path.set_extension(CATALOG_FILE_EXTENSION);
+        Self(path)
+    }
+}
+
+impl Deref for CatalogFilePath {
+    type Target = Path;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl AsRef<Path> for CatalogFilePath {
+    fn as_ref(&self) -> &Path {
+        &self.0
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ParquetFilePath(PathBuf);
+
+impl ParquetFilePath {
+    pub fn new(
+        prefix: impl Into<PathBuf>,
+        db_name: &str,
+        table_name: &str,
+        year: u16,
+        month: u8,
+        day: u8,
+        file_number: usize,
+    ) -> Self {
+        let mut path = prefix.into();
+        path.push("dbs");
+        path.push(db_name);
+        path.push(table_name);
+        path.push(format!("{year}-{month:02}-{day:02}"));
```

Comment:
If we use `min_time` as an argument, probably better to use a `strtime` equivalent format string here. Think we'll need to use Chrono for this?

Reply:
These days it or time are both maintained now, but I think to keep it consistent I'll use chrono